### PR TITLE
Macos support

### PR DESCRIPTION
**Pull request by sdk-banuba**

### CODE REVIEW

File: camera/macos/camera.mm [hunk 1]
```diff
@@ -229,7 +229,6 @@ - (void)processFrame:(CMSampleBufferRef)sampleBuffer
 {
     if (m_impl->wrapped != nullptr) {
         [m_impl->wrapped stop];
-        [m_impl->wrapped release];
```
Comment (dmitrystoyanovsky) [inline camera/macos/camera.mm:232]: Why was this removed?

File: interfaces/effect_player.hpp [hunk 2]
```diff
@@ -117,6 +117,17 @@ namespace bnb::oep::interfaces
          * @example draw()
          */
         virtual void draw() = 0;
+
```
Comment (dmitrystoyanovsky) [inline interfaces/effect_player.hpp:120]: This should be under define and activated only in the case of Metal - else all other samples depending on that interface will be broken.

File: interfaces/offscreen_render_target.hpp [hunk 3]
```diff
@@ -120,6 +120,17 @@ namespace bnb::oep::interfaces
          * @example get_current_buffer_texture()
          */
         virtual rendered_texture_t get_current_buffer_texture() = 0;
+
+        /**
```
Comment (dmitrystoyanovsky) [inline interfaces/offscreen_render_target.hpp:124]: The same as above. Please review the interfaces before doing something. In general it wasn't expected that these files will be ever changed.

**Review by dmitrystoyanovsky — CHANGES_REQUESTED** [review]

File: offscreen_effect_player/offscreen_effect_player.cpp [hunk 4]
```diff
@@ -43,9 +44,10 @@ namespace bnb::oep
     /* offscreen_effect_player::~offscreen_effect_player */
     offscreen_effect_player::~offscreen_effect_player()
     {
-        m_ep->surface_destroyed();
-        // Deinitialize offscreen render target, should be performed on render thread.
+        // Switches playback state to inactive state and deinitializes offscreen render target.
```
Comment (dmitrystoyanovsky) [inline offscreen_effect_player/offscreen_effect_player.cpp:47]: Fix comment wording - unclear

**Review by dmitrystoyanovsky — CHANGES_REQUESTED** [review]

File: offscreen_effect_player/offscreen_effect_player.cpp [hunk 5]
```diff
@@ -43,9 +44,10 @@ namespace bnb::oep
...
+        // Must be performed on render thread.
         auto task = [this]() {
+            m_ep->surface_destroyed();
```
Comment (dmitrystoyanovsky) [inline offscreen_effect_player/offscreen_effect_player.cpp:50]: Why is the surface_destroyed should be called in the render thread?

**Review by dmitrystoyanovsky — CHANGES_REQUESTED** [review]

File: offscreen_effect_player/offscreen_effect_player.cpp [hunk 6]
```diff
@@ -24,6 +24,7 @@ namespace bnb::oep
             m_ort->init(width, height);
             m_ort->activate_context();
             m_ep->surface_created(width, height);
+            m_ep->set_render_surface(m_ort->get_layer());
```
Comment (dmitrystoyanovsky) [inline offscreen_effect_player/offscreen_effect_player.cpp:27]: This also should be under define

**Review by dmitrystoyanovsky — CHANGES_REQUESTED** [review]

File: camera/macos/camera.mm (hunk 1 above)
Comment (sdk-banuba) [reply camera/macos/camera.mm:232]: When I was trying to use the camera module in OEP-macos, I ran into the error: «ARC forbids explicit message send of «release»». But now I don’t use the camera module, so, if you think that this line is needed, I can return it back.

File: offscreen_effect_player/offscreen_effect_player.cpp (hunk 5 above)
Comment (sdk-banuba) [reply offscreen_effect_player/offscreen_effect_player.cpp:50]: Because this is states in the description of the bnb::interfaces::effect_player::surface_destroyed(). 
There were freezes of the app (during exit) until this change.

File: camera/macos/camera.mm (hunk 1 above)
Comment (dmitrystoyanovsky) [reply camera/macos/camera.mm:232]: Based on your statement ARC should be enabled for the module if this file was included in the configuration. Think about the consequences of your change if ARC is not enabled for the project.

File: interfaces/effect_player.hpp (hunk 2 above)
Comment (dmitrystoyanovsky) [reply interfaces/effect_player.hpp:120]: Better remove this change to interfaces and move it to the sample as a specific.

File: offscreen_effect_player/offscreen_effect_player.cpp (hunk 6 above)
Comment (dmitrystoyanovsky) [reply offscreen_effect_player/offscreen_effect_player.cpp:29]: Check style - as far as I remember Banuba SDK assumes #ifdef at the beginning of the line, without indentation.

Comment (dmitrystoyanovsky) [reply offscreen_effect_player/offscreen_effect_player.cpp:29]: Again the implementation of Metal is in the sample, so this specific should be moved to the sample.

File: offscreen_effect_player/offscreen_effect_player.cpp (hunk 5 above)
Comment (dmitrystoyanovsky) [reply offscreen_effect_player/offscreen_effect_player.cpp:54]: OK

**Review by dmitrystoyanovsky — CHANGES_REQUESTED** [review]

File: camera/macos/camera.mm (hunk 1 above)
Comment (dmitrystoyanovsky) [reply camera/macos/camera.mm:232]: Update module config to enable ARC always.

**Review by dmitrystoyanovsky — CHANGES_REQUESTED** [review]

**Review by vadimVoloshanov — APPROVED** [review]
